src/test-metrics.jl

```diff
@@ -38,11 +38,11 @@ function gather_test_metrics(t::AsyncTestSuite)
     if !t.disabled
         for sub_testset in t.sub_testsets
             gather_test_metrics(sub_testset)
-            combine_test_metrics(t, sub_testset)
+            t = combine_test_metrics(t, sub_testset)
```

Comment:
```suggestion
            combine_test_metrics(t, sub_testset)
```

This change is not needed, as we always mutate `t::AsyncTestSuite`.

Reply:
We do need it. if you remember this gives more flexibility as to what returns from the overiden `combine_test_metrics`

for example, in some functions i was returning the child itself (after mutating it).

Reply:
This doesn't apply to the case that the first argument is a `AsyncTestSuite`. If you truly want to support this, then `gather_test_metrics` should also return `t` in here (and then you need to make sure that the returned `t` from `gather_test_metrics` is used appropriately at each of its call-sites). For now, adding an assertion might be a good idea:

```suggestion
            sub_t = combine_test_metrics(t, sub_testset)
            @assert t === sub_t
```

Reply:
shouldn't [these lines](https://github.com/RelationalAI-oss/XUnit.jl/blob/4ae90eaf786f2a3ef8b584c3eafea0bb69023fd9/src/test-metrics.jl#L116) also be included in this discussion?

Reply:
`combine_test_metrics` is always supposed to return the `parent`. We're not special-casing that.